src/Utilities/QuaternionHelpers.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines helper functions for converting between `std::array` and boost
+/// quaternions.
+
+#pragma once
+
+#include <array>
+#include <boost/math/quaternion.hpp>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+/// \brief Convert a `boost::math::quaternion` to a `std::array<DataVector, 1>`
+///
+/// \details Functions of time return `std::array<DataVector, N>`s from their
+/// `func` members where `N` is the number of derivatives of the function.
+/// For quaternions, only the quaternion itself is stored so only
+/// need to create a `std::array<DataVector, 1>` to return
+template <typename T>
+std::array<DataVector, 1> quaternion_to_array_dv(
+    const boost::math::quaternion<T>& input) noexcept {
```

Comment:
If the result is being stored in a DataVector then `T` must be `double`.

Reply:
The quaternion *could* be of any type (I don't know why it would be anything other than a `double`, but it could be), so I just `static_cast<double>` for safety

Reply:
We shouldn't allow an implicit lossy conversion.  If someone really wants to use one of the other precisions, the different Boost quaternion types have converting constructors from one another, so the user can cast to/from the one they want.

Reply:
Ok I'll just get rid of the template all together